### ISSUE TITLE
Fly links plugin

### ISSUE THREAD

**Issue by RobyB** [jonatkins/ingress-intel-total-conversion#736]
Just a question to the developer of this plugin.
It would be really nice being able to select a certain region on the map on which calculating links/fields...
Probably simplest thing would be delimiting the area by a polyline with draw tools, rather than clicking on all interested portals...
Then maybe with CTRL click you could remove any unwanted portal from the selected region...
Example of portals you would like to skip are:
- Enemy portals
- Not so easily accessible portals
- Already linked portals (this is more tricky to implement i guess)

**Comment by Hackswell** [comment]
I like Asnapper's implementation.  Works well! It'd be easier with a polygon disable/enable, but still much better than before!  Great Job!

Any way to get his patch to fly-links put in upstream?  And his plugin listed on the page?

**Comment by asnapper** [comment]
I don't think that "dirty hack" is worth for upstream ;)

**Comment by Hackswell** [comment]
I've seen nothing better added in over a year. ;)  Once it's in, it'll get the proper attention it needs.  Just my US$0.03.

**Comment by asnapper** [comment]
Thanks. Forgive me that i'm not up to date when it comes to iitc (I
basically stopped playing ingress), but isn't there any other plugin which
allows you to create a an editable list of portals? Because i think this is
a feature that many other plugins use/could use, so it would probably be
better to make it a bit more generic.

**Comment by Hackswell** [comment]
I agree. Probably be better to make an addition to draw tools to "select" all portals within a closed polygon or touching a multi-line.  Then any plugin could set a menu item for actions on the "selection".

**Comment by Yannisiman** [comment]
chrislennon  made the polygon selection fly links plugin.

https://github.com/chrislennon/IITCPlugins

**Comment by Seraphli** [comment]
I manage to update chrislennon's tweak and implement ray cast algorithm.
Here is an example.
![](https://github.com/Seraphli/IITCPlugins/raw/master/polygon-selection.png)
https://github.com/Seraphli/IITCPlugins
